cluster-autoscaler/core/scaleup/orchestrator/executor.go

```diff
@@ -141,14 +144,21 @@ func (e *scaleUpExecutor) executeScaleUp(
 	nodeInfo *schedulerframework.NodeInfo,
 	availableGPUTypes map[string]struct{},
 	now time.Time,
+	allOrNothing bool,
 ) errors.AutoscalerError {
 	gpuConfig := e.autoscalingContext.CloudProvider.GetNodeGpuConfig(nodeInfo.Node())
 	gpuResourceName, gpuType := gpu.GetGpuInfoForMetrics(gpuConfig, availableGPUTypes, nodeInfo.Node(), nil)
 	klog.V(0).Infof("Scale-up: setting group %s size to %d", info.Group.Id(), info.NewSize)
 	e.autoscalingContext.LogRecorder.Eventf(apiv1.EventTypeNormal, "ScaledUpGroup",
 		"Scale-up: setting group %s size to %d instead of %d (max: %d)", info.Group.Id(), info.NewSize, info.CurrentSize, info.MaxSize)
 	increase := info.NewSize - info.CurrentSize
-	if err := info.Group.IncreaseSize(increase); err != nil {
+	var err error
+	if allOrNothing {
+		err = info.Group.AtomicIncreaseSize(increase)
+	} else {
+		err = info.Group.IncreaseSize(increase)
+	}
+	if err != nil {
```

Comment:
Also, this should fallback to `IncreaseSize(increase)` if `AtomicIncreaseSize(increase)` is not implemented.